aws/service_v2.go

```diff
@@ -1250,3 +1325,65 @@ func (j *ExponentialJitterBackoff) BackoffDelay(attempt int, err error) (time.Du
 
 	return retryTime, nil
 }
+
+// SupportedRegionsForClient list outs the valid regions for a service based on service id
+func GetSupportedRegionsForClient(ctx context.Context, d *plugin.QueryData, serviceId string) ([]string, error) {
+	var partitionName string
+	var partition endpoints.Partition
+
+	// If valid regions list is already avialable in cache, return it
+	cacheKey := fmt.Sprintf("supported-regions-%s", serviceId)
+	if cachedData, ok := d.ConnectionManager.Cache.Get(cacheKey); ok {
+		return cachedData.([]string), nil
+	}
+
+	// Get the partition of the AWS account plugin is connected to
+	if cachedData, ok := d.ConnectionManager.Cache.Get("getAccountPartition"); ok {
+		partitionName = cachedData.(string)
+	}
+
+	// If partition name is not available,
+	// try to fetch partiton from APIs
```

Comment:
```suggestion
	// If partition name is not available, try to fetch it from APIs
```

Also, which APIs specifically are we using when trying to get the partition name? The comment isn't very specific